### DISCUSSION TITLE
min number of request generated by the schematheiss.

### DISCUSSION THREAD

**Discussion by hegebharat** [schemathesis/schemathesis#1104] [General]
We are testing some URL. For the particular POST method no of requested generated was 1.  We tested  multiple time,
this behavior observed few time. So I have below questions.

1) Can we set the min request generation ? OR can user control this behavior?
2) what factor number of request generation depends on? Is there any way to calculate this?

Thanks a lot for the answers.

**Comment by Stranger6667** [comment]
Hi!

Thank you for asking! 

> Can we set the min request generation ? OR can user control this behavior?

There is no way to control the minimum number of generated test cases. It mostly comes from the underlying Hypothesis workflows.

> what factor number of request generation depends on?

The actual number depends on:
- Schema constraints. For example, there are only 2 possible values for `{"type": "boolean"}` and Hypothesis stops when all possibilities are exhausted;
- Whether a failure was found. Once a failure is found, Hypothesis minimizes it (if the `shrinking` phase is enabled) and verifies it by re-running it multiple times (it may detect if it is a flaky failure) and stops execution;
- The testing process is randomized and depends on the `seed` value, which means that Hypothesis may find a failure at a random step of the execution, then it stops execution as described in the previous point;
- The value of the `max_examples` setting. Once these many examples are generated without finding a failing one, execution will terminate (there are some more details about this cap under the hood, but it is more relevant to Hypothesis behavior in some corner-cases like very complex strategies & how Hypothesis's health checks work)
- Enabled execution phases. For e.g. the `shrinking` phase tries to minify a failing example before reporting it.
- Internal Hypothesis health checks also affect the total number of executed requests

> Is there any way to calculate this?

I don't think it is possible to calculate an exact number, only a possible range with some confidence. However, once https://github.com/schemathesis/schemathesis/issues/1081 is implemented it will be easier to calculate the upper bound when:
- this feature is enabled (there will be no flakiness detection)
- disabled shrining phase

**Reply by hegebharat** [reply]
Hello @Stranger6667 ,

Thanks a lot for the detailed answer.

**Reply by Stranger6667** [reply]
You're welcome, @hegebharat :)
